prettier.go

```diff
@@ -0,0 +1,170 @@
+package metricsql
+
+import (
+	"fmt"
+	"strings"
+)
+
+const indentString = "  "
+
+func paddings(indent int) string {
+	var str strings.Builder
+	for i := 0; i < indent; i++ {
+		str.WriteString(indentString)
+	}
+	return str.String()
+}
+
+func shouldWrap(expr Expr) bool {
+	switch expr.(type) {
+	case *MetricExpr, *NumberExpr, *StringExpr, *FuncExpr:
+		return false
+	default:
+		return true
+	}
+}
+
+func wrapWithBraces(expr Expr, b *strings.Builder, indent, maxLineLength int) {
+	paddings := paddings(indent)
+	if shouldWrap(expr) {
+		b.WriteString(paddings + "(\n")
+		b.WriteString(prettier(expr, indent+1, maxLineLength))
+		b.WriteString("\n" + paddings + ")")
+	} else {
+		b.WriteString(prettier(expr, indent, maxLineLength))
+	}
+}
+
+func prettier(expr Expr, indent, maxLineLength int) string {
+	var str strings.Builder
+	var b []byte
+	b = expr.AppendString(b)
+	if len(b) <= maxLineLength {
+		paddings := paddings(indent)
+		str.WriteString(paddings)
+		str.Write(b)
+		return str.String()
+	}
+	switch e := expr.(type) {
+	case *BinaryOpExpr:
+		var binaryOpStr strings.Builder
+
+		wrapWithBraces(e.Left, &binaryOpStr, indent+1, maxLineLength)
+		buildBinaryOp(&binaryOpStr, e, indent)
+		wrapWithBraces(e.Right, &binaryOpStr, indent+1, maxLineLength)
+
+		str.WriteString(binaryOpStr.String())
+	case *RollupExpr:
+		var rollupStr strings.Builder
+
+		wrapWithBraces(e.Expr, &rollupStr, indent, maxLineLength)
+		buildRollupFunc(&rollupStr, e)
+
+		str.WriteString(rollupStr.String())
+	case *AggrFuncExpr:
+		var aggrFuncStr strings.Builder
+
+		buildAggrFuncString(&aggrFuncStr, e, indent, maxLineLength)
+
+		str.WriteString(aggrFuncStr.String())
+	case *MetricExpr, *NumberExpr, *StringExpr:
+		paddings := paddings(indent)
+		var b []byte
+		str.WriteString(paddings)
+		str.Write(e.AppendString(b))
+	case *FuncExpr:
+		buildFuncExpr(&str, e, indent, maxLineLength)
+	}
```

Comment:
Wdyt about adding `default` for this switch so that possible new tokens will also work with this function?